app-misc/elasticsearch/files/elasticsearch-env.patch

```diff
@@ -0,0 +1,18 @@
+Since ES 8.0.0 upstream stopped distributing a 'no-JDK' package.
+This requires us to set `ES_JAVA_HOME` else it fails to find the
+JDK that we unbundled and will not start.
+--- a/bin/elasticsearch-env
++++ b/bin/elasticsearch-env
+@@ -36,6 +36,12 @@ ES_HOME=`dirname "$ES_HOME"`
+ ES_CLASSPATH="$ES_HOME/lib/*"
+ LAUNCHERS_CLASSPATH="$ES_CLASSPATH:$ES_HOME/lib/launchers/*"
+ 
++# Set our JVM in a Gentoo-specific manner
++if [ -z "$ES_JAVA_HOME" ]; then
++  GENTOO_VM=$(depend-java-query --get-vm virtual/jre:17)
++  ES_JAVA_HOME=$(java-config-2 --select-vm=${GENTOO_VM} --jre-home)
```

Comment:
I just checked the behavior of `depend-java-query` and `java-config-2`. The bottom line is that you want to test $GENTOO_VM for non-emptiness before calling `java-config-2`, as if $GENTOO_VM is empty, then `java-config-2` will use "some" JVM (probably the system JVM, haven't checked).

Reply:
Looks like it falls back to the system VM after some quick behaviour testing, (though that may just also be the first VM that it lists...).

Added a conditional to not set ES_JAVA_HOME if we don't get a JRE and echo a warning instead.